webapp/src/workspace.ts

```diff
@@ -139,7 +118,11 @@ async function switchToMemoryWorkspace(reason: string): Promise<void> {
 
 export function getHeaders(withDeleted = false) {
     maybeSyncHeadersAsync().done();
-    let r = allScripts.map(e => e.header).filter(h => (withDeleted || !h.isDeleted) && !h.isBackup)
+    const cloudUserId = auth.user()?.id;
+    let r = allScripts.map(e => e.header).filter(h =>
+        (withDeleted || !h.isDeleted) &&
+        !h.isBackup &&
+        (!h.cloudUserId || h.cloudUserId === cloudUserId))
     r.sort((a, b) => b.recentUse - a.recentUse)
     return r
 }
```

Comment:
Some thoughts on merging two workspaces (instead of layer cloud on top of workspace-agnostic layer):
- This was the path Sam and Peli took